1.20.4/src/main/java/ru/vidtu/ias/screen/AccountScreen.java

```diff
@@ -0,0 +1,328 @@
+/*
+ * In-Game Account Switcher is a mod for Minecraft that allows you to change your logged in account in-game, without restarting Minecraft.
+ * Copyright (C) 2015-2022 The_Fireplace
+ * Copyright (C) 2021-2024 VidTu
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>
+ */
+
+package ru.vidtu.ias.screen;
+
+import net.minecraft.ChatFormatting;
+import net.minecraft.Util;
+import net.minecraft.client.gui.GuiGraphics;
+import net.minecraft.client.gui.components.Button;
+import net.minecraft.client.gui.components.EditBox;
+import net.minecraft.client.gui.components.PlayerSkinWidget;
+import net.minecraft.client.gui.components.Tooltip;
+import net.minecraft.client.gui.navigation.CommonInputs;
+import net.minecraft.client.gui.screens.AlertScreen;
+import net.minecraft.client.gui.screens.Screen;
+import net.minecraft.client.resources.DefaultPlayerSkin;
+import net.minecraft.network.chat.CommonComponents;
+import net.minecraft.network.chat.Component;
+import org.lwjgl.glfw.GLFW;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import ru.vidtu.ias.IAS;
+import ru.vidtu.ias.account.Account;
+import ru.vidtu.ias.config.IASStorage;
+
+public final class AccountScreen extends Screen {
+    /**
+     * Logger for this class.
+     */
+    private static final Logger LOGGER = LoggerFactory.getLogger("IAS/AccountScreen");
+
+    /**
+     * Parent screen, {@code null} if none.
+     */
+    private final Screen parent;
+
+    /**
+     * Search widget.
+     */
+    private EditBox search;
+
+    /**
+     * Account list widget.
+     */
+    private AccountList list;
+
+    /**
+     * Player skin widget.
+     */
+    private PlayerSkinWidget skin;
+
+    /**
+     * Login button.
+     */
+    private Button login;
+
+    /**
+     * Offline login button.
+     */
+    private Button offlineLogin;
+
+    /**
+     * Edit button.
+     */
+    private Button edit;
+
+    /**
+     * Edit button.
+     */
+    private Button delete;
+
+    /**
+     * Creates a new screen.
+     *
+     * @param parent Parent screen, {@code null} if none
+     */
+    public AccountScreen(Screen parent) {
+        super(Component.translatable("ias.accounts"));
+        this.parent = parent;
+    }
+
+    @Override
+    protected void init() {
+        // Bruh.
+        assert this.minecraft != null;
```

Comment:
lol nice

Reply:
intellij kept bragging about nullability, i didn't want to disable control flow checks, suppress them for every statement where minecraft field is used, or use preconditions or similar checks that will actually check field at runtime. i came up with this dirty hack. looking at mc code, init should always be called after assigning minecraft field to mc singleton and if some mods will break this (they shouldn't, this will break practically any GUI), the NPE thrown will clearly indicate that (thanks to java 14 detailed NPEs)